deepcell/graph_utils.py

```diff
@@ -0,0 +1,274 @@
+
+    return feature_matrix
+
+
+def get_celltypes(label_image, celltype_image):
+    """Query cell type image with a given label image and get cell
+    type for each cell
+
+    Args:
+        label_image (np.array): Label image where every cell has been given a
+            unique integer id
+        celltype_image (np.array): Image where pixels belonging to each cell
+            are given an integer id specifying cell type
+
+    Raises:
+        ValueError: if image and label_image bach does not match
+
+    Returns:
+        np.array(float): Cell type matrix consisting of cell types for
+            each cell
+    """
+    if label_image.shape[0] != celltype_image.shape[0]:
+        raise ValueError("label_image and cell_type_image must have same size for "
+                         " the batch dimension")
+
+    # Find out max number of cells
+    max_cells = get_max_cells(label_image)
+
+    # Instantiate cell type matrix
+    celltype_matrix = np.zeros((label_image.shape[0], max_cells))
+
+    for batch in range(label_image.shape[0]):
+        label_image_batch = np.squeeze(label_image[batch])
+        celltype_image_batch = np.squeeze(celltype_image[batch])
+
+        props = regionprops_table(label_image_batch,
+                                  properties=['coords', 'label'])
+
+        for i in range(len(props['coords'])):
+            coords = props['coords'][i]
+            cell_type_list = celltype_image_batch[coords[:, 0], coords[:, 1]]
+            cell_type = mode(cell_type_list, axis=None).mode[0]
+            celltype_matrix[batch, i] = cell_type
+
+    return celltype_matrix
+
+
+def adj_to_degree(adj, power=-0.5, epsilon=1e-5):
+    """ Convert adjacency matrix to degree matrix
+
+    Args:
+        adj (float): Adjacency matrix
+        power (float): tbd
+        epsilon (float): factor to fix float error
+
+    Returns:
+        np.array: Degree matrix raised to a given power
+    """
+    if not isinstance(adj, sp.csr.csr_matrix):
+        adj = sp.csr.csr_matrix(adj)
+
+    # adj is (batch, row, col)
+    degrees = np.sum(adj, axis=1)  # this should be (batch, col)
+
+    degree_matrix = np.zeros(adj.shape)
+    for batch, degree in enumerate(degrees):
+        if power is not None:
+            degree = (degree + epsilon) ** power
+        degree_matrix[batch] = np.diagflat(degree)
+
+    return degree_matrix
+
+
+def normalize_adj(adj):
+    """Symmetrically normalize adjacency matrix.
+
+    Args:
+        adj (float): Adjacency matrix
+
+    Returns:
+        np.array: Symmetrically normalized adjacency matrix in
+            coordinate format
+    """
+    d_mat_inv_sqrt = adj_to_degree(adj, power=-0.5)
+
+    # Compute D^-0.5AD^0.5 - Recall that D and A are symmetric
+    normalized_adjacency = np.matmul(d_mat_inv_sqrt, adj)
+    normalized_adjacency = np.matmul(normalized_adjacency, d_mat_inv_sqrt)
+
+    return normalized_adjacency
+
+
+def preprocess_adj(adj):
```

Comment:
Can we rename this function to be more descriptive? Maybe something like `convert_matrix_to_tuple`? Ideally it will be easy to know what's happening to the data from the function name.